docs/src/getting_started.md

```diff
@@ -1,62 +1,121 @@
 
-First, import the module like this
+This guide will introduce you to the basics of using classification trees with the `DecisionTreeAndRandomForest.jl` package.
+
+## What is a Decision Tree?
+
+A decision tree is a machine learning model used for both classification and regression tasks. It uses a tree-like structure where internal nodes represent features, branches represent decision rules, and each leaf node represents an outcome. Decision trees are easy to interpret and visualize, making them popular for many applications.
+
+## What is a Random Forest?
+
+A random forest is an ensemble learning method that constructs multiple decision trees during training and outputs the mode of the classes (classification) or mean prediction (regression) of the individual trees. This approach improves accuracy and reduces overfitting.
+
+
+## Overview of Features
+
+
+- **Classification Trees**: Build trees for classifying data.
+- **Regression Trees**: Construct trees for predicting continuous values.
+- **Random Forests**: Ensemble method that combines multiple decision trees to improve accuracy and robustness.
+- **Custom Splitting Criteria**: Support for various splitting criteria such as Gini Impurity, Information Gain, and Variance Reduction.
+
+
+## Basic Example of a Classification Tree
+
+### Step 1: Import the Module
+
+First, import the `DecisionTreeAndRandomForest` module:
 
 ```@example 1
 using DecisionTreeAndRandomForest
 ```
 
-Second we need some training data and their respective labels.
+### Step 2: Prepare Training Data
+
+Prepare some training data and their respective labels:
 
 ```@example 1
+# Prepare some training data and their respective labels:
 data = ["dog" 37.0; "dog" 38.4; "dog" 40.2; "dog" 38.9; "human" 36.2; "human" 37.4; "human" 38.8; "human" 36.2]
 labels = ["healthy", "healthy", "sick", "healthy", "healthy", "sick", "sick", "healthy"]
 nothing # hide
 ```
 
-After that we can initialiate a tree. There are two constructors:
-1. One only takes the data and labels as parameters.
-2. The other one can also take values for `max_depth` and `min_samples_split`.
+### Step 3: Initialize a Tree
+
+Initialize a classification tree. There are two constructors:
+1. One that only takes the data and labels as parameters.
+2. Another that also takes values for `max_depth` and `min_samples_split`.
 
 ```@example 1
 tree = ClassificationTree(data, labels, split_gini)
 other_tree = ClassificationTree(3, 2, split_gini, data, labels)
 nothing # hide
 ```
 
-We can build the tree using the `fit` function.
+### Step 4: Build the Tree
 
-!!! info
-    At the moment the `fit` function uses the Gini-Impurity to find the optimal split. In the future you can provide a custom function by passing it into the `fit` function.
+Build the tree using the `fit` function.
 
+!!! info
+    Currently, the `fit` function uses the Gini-Impurity to find the optimal split. In the future, you can provide a custom function by passing it into the `fit` function.
```

Comment:
this is outdated. We can now use different splitting criterion for the fit function